torch/linalg/__init__.py

```diff
@@ -431,6 +431,55 @@
             [1, 2, 2, 2]])
 """)
 
+multi_dot = _add_docstr(_linalg.linalg_multi_dot, r"""
+linalg.multi_dot(tensors, *, out=None)
+
+Efficiently multiplies two or more matrices given by :attr:`tensors` by
+selecting the matrix chain multiplication order that results in the least
+number of arithmetic operations.
```

Comment:
Overall these docs seem really good to me and I wouldn't want to hold the PR on further improvements, but it might be interesting to think if there's an example we can provide here of how the reordering works. I like NumPy's multi_dot note on the topic, for example.